src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -389,6 +451,74 @@ private static string GetEscapedParameterName(AttributeData attribute, string pa
         }
     }
 
+    private static bool TryGetAsParametersConstructor(INamedTypeSymbol type, out bool? isParameterlessConstructor, [NotNullWhen(true)] out IEnumerable<ConstructorParameter>? matchedParameters)
+    {
+        isParameterlessConstructor = null;
+        matchedParameters = null;
+        if (type.IsAbstract)
+        {
+            return false;
+        }
+
+        var constructors = type.Constructors.Where(constructor => constructor.DeclaredAccessibility == Accessibility.Public && !constructor.IsStatic);
+        var numOfConstructors = constructors.Count();
+        // When leveraging parameterless constructors, we want to ensure we only emit for writable
+        // properties. We do not have this constraint if we are leveraging a parameterized constructor.
+        var properties = type.GetMembers().OfType<IPropertySymbol>().Where(property => property.DeclaredAccessibility == Accessibility.Public);
+        var writableProperties = properties.Where(property => !property.IsReadOnly);
+
+        if (numOfConstructors == 1)
+        {
+            var targetConstructor = constructors.SingleOrDefault();
+            var lookupTable = new Dictionary<ParameterLookupKey, IPropertySymbol>();
+            foreach (var property in properties)
+            {
+                lookupTable.Add(new ParameterLookupKey(property.Name, property.Type), property);
```

Comment:
I think indexers can have the same name.  Maybe you meant to filter those out above?

Reply:
Good question that led me down an interesting rabbit hole!

I'm assuming that you're referring to indexers on types like in the case below:

```
public class ClassWithIndexer
{
    private int[] arr = new int[100];
    private Dictionary<string, int> map = new Dictionary<string, int>();
    public int this[int i]
    {
        get { return arr[i]; }
        set { arr[i] = value; }
    }

    public int this[string i]
    {
        get { return map[i]; }
        set { map[i] = value; }
    }
}
```

Looking to RDF as guidance, we observe two problems here:

It looks like the `AsParameters` implementation in RDF isn't supporting indexers at all. Amongst other things, it produces incorrect code including initializations like the following:

```
$args_local = .New Microsoft.AspNetCore.Http.Generators.Tests.ClassWithIndexer(){
            Item = $Item_local
        };
```

which disregard that the indexer cannot be initialized with an object initializer. It seems like support for indexers on AsParameters-enabled types wasn't considered as part of the original proposal so we'll have to think through how the implementation will work in both RDF and RDG. I've filed https://github.com/dotnet/aspnetcore/issues/48118 to track this.

Second is with regards to the scenario where multiple indexers are defined (both which emit the underlying `Item` property name). RDF has support for this in its argument checking. It'll through an exception at startup warning that duplicate argument names where found:

> ArgumentException: An item with the same key has already been added. Key: Item

This is smelly code though because we're just throwing the unhandled exception from attempting to add to [FactoryContext.TrackedParameters](https://source.dot.net/#Microsoft.AspNetCore.Http.Extensions/RequestDelegateFactoryContext.cs,42) without informing the user why.

I'll noodle on how to handle this more elegantly in RDG but may approach this in a follow-up when we migrate the [RequestDelegateThrowsWhenParameterNameConflicts](https://github.com/dotnet/aspnetcore/blob/0a7d70fc004572cfaaaa5938fd82b80c99bf1990/src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs#L3160) test.

Reply:
Maybe we should just skip properties with parameters for now?